--- ./ld/ld.c.orig	2007-09-01 13:24:22.000000000 -0700
+++ ./ld/ld.c	2007-09-01 13:24:22.000000000 -0700
@@ -81,6 +81,8 @@
 #include "layout.h"
 #include "pass2.h"
 
+#include <config.h>
+
 /* name of this program as executed (argv[0]) */
 __private_extern__ char *progname = NULL;
 /* indication of an error set in error(), for processing a number of errors
@@ -1871,6 +1873,11 @@
 	else{
 	    next_root = p;
 	}
+       if(next_root == NULL) {
+#ifdef CROSS_SYSROOT
+         next_root = CROSS_SYSROOT;
+#endif
+       }
 	if(next_root != NULL){
 	    for(i = 0; standard_dirs[i] != NULL; i++){
 		p = allocate(strlen(next_root) +
--- ./misc/libtool.c.orig	2007-09-01 13:24:22.000000000 -0700
+++ ./misc/libtool.c	2007-09-01 13:24:22.000000000 -0700
@@ -55,6 +55,8 @@
 #include <servers/bootstrap.h>
 #endif
 
+#include <config.h>
+
 /* used by error routines as the name of the program */
 char *progname = NULL;
 
@@ -993,6 +995,11 @@
 	else{
 	    next_root = getenv("NEXT_ROOT");
 	}
+	if(next_root == NULL) {
+#ifdef CROSS_SYSROOT
+          next_root = CROSS_SYSROOT;
+#endif
+	}
 	if(next_root != NULL){
 	    for(i = 0; standard_dirs[i] != NULL; i++){
 		p = allocate(strlen(next_root) +
