--- ./app.c.orig	2007-09-01 13:24:11.000000000 -0700
+++ ./app.c	2007-09-01 13:24:21.000000000 -0700
@@ -29,6 +29,10 @@
 #include "app.h"
 #include "messages.h"
 
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
 FILE *scrub_file = NULL;
 char *scrub_string = NULL;
 char *scrub_last_string = NULL;
--- ./input-file.c.orig	2007-09-01 13:24:11.000000000 -0700
+++ ./input-file.c	2007-09-01 13:24:21.000000000 -0700
@@ -37,6 +37,10 @@
 #include "app.h"
 #include "as.h"
 
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
 /*
  * This variable is non-zero if the file currently being read should be
  * preprocessed by app.  It is zero if the file can be read straight in.
--- ./read.c.orig	2007-09-01 13:24:11.000000000 -0700
+++ ./read.c	2007-09-01 13:24:21.000000000 -0700
@@ -60,6 +60,10 @@
 #include "i386.h"
 #endif
 
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
 /*
  * Parsing of input is done off of this pointer which points to the next char
  * of source file to parse.
