--- boolean.h	2006-09-11 19:16:13.000000000 -0700
+++ boolean.h	2007-09-01 11:55:42.000000000 -0700
@@ -27,8 +27,10 @@
 #include "mach/ppc/boolean.h"
 #elif defined (__i386__) || defined(__x86_64__)
 #include "mach/i386/boolean.h"
+#elif defined(__arm__)
+#include <mach/arm/boolean.h>
 #else
 #error architecture not supported
 #endif
 
 #endif /* _MACH_MACHINE_BOOLEAN_H_ */
--- exception.h	2006-09-11 19:16:13.000000000 -0700
+++ exception.h	2007-09-01 11:55:42.000000000 -0700
@@ -27,6 +27,8 @@
 #include "mach/ppc/exception.h"
 #elif defined (__i386__) || defined(__x86_64__)
 #include "mach/i386/exception.h"
+#elif defined(__arm__)
+#include "mach/arm/exception.h"
 #else
 #error architecture not supported
 #endif
--- kern_return.h	2006-09-11 19:16:13.000000000 -0700
+++ kern_return.h	2007-09-01 11:55:42.000000000 -0700
@@ -27,6 +27,8 @@
 #include "mach/ppc/kern_return.h"
 #elif defined (__i386__) || defined(__x86_64__)
 #include "mach/i386/kern_return.h"
+#elif defined(__arm__)
+#include "mach/arm/kern_return.h"
 #else
 #error architecture not supported
 #endif
--- processor_info.h	2006-09-11 19:16:13.000000000 -0700
+++ processor_info.h	2007-09-01 11:55:42.000000000 -0700
@@ -27,6 +27,8 @@
 #include "mach/ppc/processor_info.h"
 #elif defined (__i386__) || defined(__x86_64__)
 #include "mach/i386/processor_info.h"
+#elif defined (__arm__)
+#include "mach/arm/processor_info.h"
 #else
 #error architecture not supported
 #endif
--- rpc.h	2006-09-11 19:16:13.000000000 -0700
+++ rpc.h	2007-09-01 11:55:42.000000000 -0700
@@ -27,6 +27,8 @@
 #include "mach/ppc/rpc.h"
 #elif defined (__i386__) || defined(__x86_64__)
 #include "mach/i386/rpc.h"
+#elif defined(__arm__)
+#include <mach/arm/rpc.h>
 #else
 #error architecture not supported
 #endif
--- thread_state.h	2006-09-11 19:16:13.000000000 -0700
+++ thread_state.h	2007-09-01 11:55:42.000000000 -0700
@@ -27,6 +27,8 @@
 #include "mach/ppc/thread_state.h"
 #elif defined (__i386__) || defined(__x86_64__)
 #include "mach/i386/thread_state.h"
+#elif defined(__arm__)
+#include "mach/arm/thread_state.h"
 #else
 #error architecture not supported
 #endif
--- thread_status.h	2006-09-11 19:16:13.000000000 -0700
+++ thread_status.h	2007-09-01 11:55:42.000000000 -0700
@@ -27,6 +27,8 @@
 #include "mach/ppc/thread_status.h"
 #elif defined (__i386__) || defined(__x86_64__)
 #include "mach/i386/thread_status.h"
+#elif defined (__arm__)
+#include "mach/arm/thread_status.h"
 #else
 #error architecture not supported
 #endif
--- vm_param.h	2006-09-11 19:16:13.000000000 -0700
+++ vm_param.h	2007-09-01 11:55:42.000000000 -0700
@@ -27,7 +27,9 @@
 #include "mach/ppc/vm_param.h"
 #elif defined (__i386__) || defined(__x86_64__)
 #include "mach/i386/vm_param.h"
+#elif defined(__arm__)
+#include "mach/arm/vm_param.h"
 #else
 #error architecture not supported
 #endif
 
--- vm_types.h	2006-09-11 19:16:13.000000000 -0700
+++ vm_types.h	2007-09-01 11:55:42.000000000 -0700
@@ -27,8 +27,10 @@
 #include "mach/ppc/vm_types.h"
 #elif defined (__i386__) || defined(__x86_64__)
 #include "mach/i386/vm_types.h"
+#elif defined(__arm__)
+#include <mach/arm/vm_types.h>
 #else
 #error architecture not supported
 #endif
 
 #endif /* _MACH_MACHINE_VM_TYPES_H_ */
