--- task.h	2006-09-11 15:38:37.000000000 -0700
+++ task.h	2007-09-01 11:55:42.000000000 -0700
@@ -28,6 +28,8 @@
 #include <mach/ppc/task.h>
 #elif defined(__ppc64__)
 #include <mach/ppc64/task.h>
+#elif defined(__arm__)
+#include <mach/arm/task.h>
 #else
 #error unknown architecture
 #endif 
--- thread_act.h	2006-09-11 15:38:37.000000000 -0700
+++ thread_act.h	2007-09-01 11:55:42.000000000 -0700
@@ -28,6 +28,8 @@
 #include <mach/ppc/thread_act.h>
 #elif defined(__ppc64__)
 #include <mach/ppc64/thread_act.h>
+#elif defined(__arm__)
+#include <mach/arm/thread_act.h>
 #else
 #error unknown architecture
 #endif 
