--- ./libtool.c.orig	2007-09-01 13:24:22.000000000 -0700
+++ ./libtool.c	2007-09-01 13:24:23.000000000 -0700
@@ -2389,7 +2389,7 @@
 	    if((archs[i].arch_flag.cputype & CPU_ARCH_ABI64) == CPU_ARCH_ABI64)
 		add_execute_list("ld64");
 	    else
-		add_execute_list("ld");
+	      add_execute_list(makestr(BINDIR, "/", LDPROG, NULL));
 	    if(narchs != 0 && cmd_flags.arch_only_flag.name == NULL)
 		add_execute_list("-arch_multiple");
 	    if(archs != NULL){
