--- ./arch.h.orig	2007-09-01 14:39:59.000000000 -0700
+++ ./arch.h	2007-09-01 14:40:00.000000000 -0700
@@ -104,6 +104,9 @@
 extern const struct arch_flag *get_arch_family_from_cputype(
     cpu_type_t cputype);
 
+__private_extern__ unsigned long get_shared_region_sz_from_flag(const struct
+    arch_flag *cputype);
+
 /*
  * get_byte_sex_from_flag() returns the byte sex of the architecture for the
  * specified cputype and cpusubtype if known.  If unknown it returns
--- ./bytesex.h.orig	2007-09-01 14:39:59.000000000 -0700
+++ ./bytesex.h	2007-09-01 14:40:47.000000000 -0700
@@ -51,6 +51,7 @@
 #include <mach/i386/thread_status.h>
 #include <mach/hppa/thread_status.h>
 #include <mach/sparc/thread_status.h>
+#include <mach/arm/thread_status.h>
 #include <mach-o/nlist.h>
 #include <mach-o/reloc.h>
 #include <mach-o/ranlib.h>
@@ -300,6 +301,10 @@
     struct ident_command *id_cmd,
     enum byte_sex target_byte_sex);
 
+extern void swap_arm_thread_state(
+    struct arm_thread_state *cpu,
+    enum byte_sex target_byte_sex);
+
 extern void swap_routines_command(
     struct routines_command *r_cmd,
     enum byte_sex target_byte_sex);
