--- ./src/Options.cpp.orig	2007-09-01 13:24:22.000000000 -0700
+++ ./src/Options.cpp	2007-09-01 13:24:22.000000000 -0700
@@ -29,7 +29,7 @@
 #include <fcntl.h>
 #include <ctype.h>
 #include <vector>
-
+#include "config.h"
 
 #include "Options.h"
 
@@ -1439,6 +1439,13 @@
 			fSDKPaths.push_back(path);
 		}
 	}
+
+	if(fSDKPaths.size() == 0) {
+#ifdef CROSS_SYSROOT
+		fSDKPaths.push_back(CROSS_SYSROOT);
+#endif
+	}
+
 	if ( addStandardLibraryDirectories ) {
 		libraryPaths.push_back("/usr/lib");
 		libraryPaths.push_back("/usr/local/lib");
